### ISSUE TITLE
Opening the {Template | Solution | Test }-repository is unintuitive

### ISSUE THREAD

**Issue by pvlov** [ls1intum/Artemis#9108]
### Is your feature request related to a problem?

When you want to view the {Template | Solution | Test }-repository inside of an Artemis course as a tutor through its builtin code editor, you are presented with this window:
<img width="717" alt="choice" src="https://github.com/user-attachments/assets/383565f4-366d-480f-93f7-0932668471fc">

The way to open the code editor is to now click on the symbol on the right of the link, which is very unintuitive since other applications e.g. GitHub, GitLab use this element to copy the link. Also without any description, there is no way to know what this button will do, as the only information provided is the fact that it is outbound. 

### Describe the solution you'd like

The `Copy URL` button should swap the logic with the small button next to the link.

### Describe alternatives you've considered

You could keep the functionality as is, but add a hover-text that says something along the lines of "Click here to open in the <insert editor name> Editor"

### Additional context

To replicate and see this window, go to a coding exercise of a course, click on `view` in the `tutor actions` bar and then scroll down until you see the different buttons for the repositories.

**Comment by pvlov** [comment]
I don't know if you consider this as a feature or bug, so I'll attach both labels for now. If this UX is by choice, I would love to hear the reasoning behind your choice of breaking the pattern Applications usually adhere to.

This _should_ be an easy change and therefore warrant the `good first issue` label but I'll let the you, the Artemis Maintainers, decide if it is applicable.

**Comment by krusche** [comment]
Thanks for your feedback. We really appreciate it!

This was done by choice because of the following reasons:

1. we have a dedicated **edit in editor** feature when you want to change the code of those repos online. With the review changes button, you can compare template vs solution repository

2. most instructors prefer to edit the repos locally in their own IDE

3. Opening the readonly online code editor is not done often

4. we want to reduce the clutter on the page with too many button

5. we want to have a consistent experience with other pages that show repositories 

@rabeatwork is currently re-designing this dialog to make it more appealing.

Is there a specific reason why you want to open those repos in the readonly code editor? If we understand this better, we can potentially come up with a better solution for you.

**Comment by pvlov** [comment]
Thanks for the quick reply! It is great to hear about Artemis's dedication to improving the UX of Artemis.

My concerns mainly stem from my experience working with Artemis as a tutor, so I can only really speak about it in that capacity. My main gripe with cloning the repository, is that I need to clone it. Depending on the Exercise, this can take no time at all e.g. some small OCaml exercise such as in the course FPV but can also take quite a lot of time (due to indexing etc.) if we consider larger exercises like the ones encounter in a course like EIST.

Here are some of the situations I have encountered in my usual workflow as a tutor:

- After a working period in my tutorial I want to talk about the different approaches my student took/could take and want to have the model solution on my other screen as a reference. I don't want to actually interact with the solution but just have it open on the side to be able to see what the instructors thought would be the best solution. Sometimes the solution even contains multiple approaches making it very handy to have as a reference.

- If a student has problems with the template or just questions about it, I want to be able to check the template quickly to be able to accurately answer the question of the student or clarify something by referencing the exact wording/code snippet in the template.

- When a student encounters problems due to tests failing, I often want to reference both the student submission as well as the test repository to see if there is an actual problem with the tests or if the student made a mistake that makes the tests behave weirdly.

- If a student asks me a question about their specific approach, it is much easier to check the implementation via Artemis on my device than physically going to the student or using some kind of screen share in online tutorials.

This annoyance is often mitigated by the fact that most instructors have the repositories cloned anyway but for tutors, that only want to quickly consume the content of the model solution or a students code, cloning and then immediately deleting the respective repository is rather unpleasant.

> "1. we have a dedicated edit in editor feature when you want to change the code of those repos online. With the review changes button, you can compare template vs solution repository"

The `Review Changes` button is only really practical to use when the `git diff` is small, for bigger task involving multiple files, a dedicated file explorer is more practical.

Regarding 2. and 3., I guess this is personal preference. 

> "4. we want to reduce the clutter on the page with too many button"

I think after [this issue about customising the `Open In` config](https://github.com/ls1intum/Artemis/issues/9091) is processed, this should not be a problem anymore and having a choice between viewing online and cloning will feel very obvious and concise.

> "5. we want to have a consistent experience with other pages that show repositories"

I have only seen the same dialogue in the `participation` tab, where the same issues apply. Are there any more pages where I could check the behaviour?

**Comment by Predixx** [comment]
Problem is still existing. I think that a good compromise could be that when you hover over the button that it shows a small label (as proposed). If you want I can fix that @krusche

**Comment by krusche** [comment]
Yes, we can implement a tooltip on hover. I am not 100% happy about the current user experience and would be happy to discuss alternatives to navigate into the read only editors more easily

**Comment by krusche** [comment]
We should soon work on improving the user experience for this page
